File: jeonghyeon/5week/예산.js
function solution(d, budget) {
    let number = 0; // 지원 가능 부서 개수
    d.sort((a, b) => a - b); // 오름차순
    for (let i = 0; i < d.length; i++) {
        if (budget >= d[i]) {
          budget -= d[i]; // 예산에서 해당 부서의 신청 금액 차감
          number++; // 지원 가능한 부서 개수 증가
        } else {
          break; // 예산을 초과하면 반복 종료
        }
      }
      return number;
}

File: minwook/5주차/예산.java
import java.util.*;
class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        Arrays.sort(d); // 오름차순 정렬 (최대 물품 지원을 위해 최솟값부터 진행)
        for(int n : d){ // 부서별 금액 접근
            budget -= n; // 예산에서 금액 감소
            if(budget >= 0) // 예산이 초과되기전까지 덧셈
                answer++;
        }
        return answer;
    }
}